source/ceylon/time/Date.ceylon

```diff
@@ -32,6 +32,15 @@ shared interface Date
 
     "Returns the [[DateRange]] between this and given [[Date]]."
     shared formal DateRange rangeTo( Date other );
+    
+    "Two `Date`s are considered equals if they have the same 
```

Comment:
Yes, thats what I mean ;) ... we are currently using `dayOfEra`, but we need a _formal_ definition of how two Date's are considered `equals` .. and like you said in your example I believe you have a better approach.